packages/macos/package_files/preinstall.sh

```diff
@@ -20,6 +21,23 @@ function check_errm
         fi
 }
 
+function check_arch
+{
+    local system_arch=$(uname -m)
+
+    if [ "$ARCH" = "intel64" ] && [ "$system_arch" = "arm64" ]; then
+        if ! arch -x86-64 zsh -c '' &> /dev/null; then
+            echo "ERROR: Rosetta is not installed. Please install it and try again."
+            exit 1
+        fi
+    elif [ "$ARCH" = "arm64" ] && [ "$system_arch" = "x86_64" ]; then
+        echo "ERROR: Incompatible architecture. Please use the ARM64 package on this system."
```

Comment:
If `$system_arch` is `x86_64` we should request to install the `intel64` package instead of the `ARM64`